receiver/factory.go

```diff
@@ -25,11 +26,11 @@ type TraceReceiverFactory interface {
 	// Type gets the type of the TraceReceiver created by this factory.
 	Type() string
 	// NewFromViper takes a viper.Viper config and creates a new TraceReceiver which uses next as the
-	// next TraceConsumer in the pipeline.
-	NewFromViper(cfg *viper.Viper, next consumer.TraceConsumer) (TraceReceiver, error)
-	// DefaultConfig returns the default configuration for TraceReceivers
+	// next TraceConsumer in the pipeline. Returning also the configuration used to create it.
+	NewFromViper(v *viper.Viper, next consumer.TraceConsumer) (receiver TraceReceiver, config interface{}, err error)
+	// DefaultConfig gets the default configuration for the TraceReceiver
 	// created by this factory.
-	DefaultConfig() *viper.Viper
+	DefaultConfig() interface{}
```

Comment:
Do we need a ValidateConfig(v *viper.Viper) error? That we can use in the config module to validate a config?

Reply:
If we want to run "validate" config by itself then yes, but, the ultimate validation is to create the receivers. We don't necessarily need to start them if we just want to validate the config: build the pipelines and terminate without starting any reception. Depending on how receivers are built we may miss some cases that require starting reception to error out, but, that is also not solved by a ValidateConfig method.

BTW, I should change from `Unmarshal` to `UnmarshalExact` so viper errors on exta-fields (typically these are typos).